### ISSUE TITLE
Bootstrap/Upgrade may fail when bringing to highstate

### ISSUE THREAD

**Issue by gdemonet** [scality/metalk8s#1952]
**Component**: salt

**What happened**:

Sometimes, the bootstrap script fails when bringing to highstate, with very terse error information:

```
----------
          ID: Deploy node bootstrap
    Function: salt.runner
        Name: state.orchestrate
      Result: False
     Comment: Runner function 'state.orchestrate' failed.
     Started: 07:25:48.318255
    Duration: 53214.966 ms
     Changes:   
              ----------
              return:
                  ----------
                  data:
                      ----------
                      bootstrap_master:
                          ----------
                          [...]
                          salt_|-Run the highstate_|-Run the highstate_|-state:
                              ----------
                              __id__:
                                  Run the highstate
                              __run_num__:
                                  4
                              __sls__:
                                  metalk8s.orchestrate.deploy_node
                              changes:
                                  ----------
                                  out:
                                      highstate
                                  ret:
                                      ----------
                                      bootstrap:
                                          False
                              comment:
                                  Run failed on minions: bootstrap
                              duration:
                                  35773.0
                              name:
                                  Run the highstate
                              result:
                                  False
                              start_time:
                                  07:26:05.712483
                  outputter:
                      highstate
                  retcode:
                      1
```

**What was expected**:

Bootstrap should not fail for other reasons than invalid configuration: it should try to be resilient to performance drops, that can happen in CI environments, or other environment-related issues.

Bootstrap being one of the few rare cases where a user *needs* to use the CLI, it should strive to provide meaningful error messages, inviting the user to act on those errors.

**Steps to reproduce**: Not obvious, appearing randomly in CI, less frequently on local Vagrant setup

**Resolution proposal** (optional):

- Investigate better output for Salt executions (orchestrates + states + modules)
- Understand this flaky failure when bringing to highstate using these better error messages
- Fix it!

**Comment by alexandre-allard** [comment]
After some investigations, it seems that the highstate goes well on the minion (no errors) and it returns the good result to the master.
But, apparently the master is not waiting enough for the answer (in the logs, the minion answered ~20 seconds after the master returned the failed status), or maybe there is another problem when it checks the job liveness.
Maybe it could be related to the restart of the minion, because the master polls every x seconds to check if the job is still running and maybe the polling arrives when the minion is restarting.
Still investigating, trying to raise some timeout values, etc.

**Comment by alexandre-allard** [comment]
So, I finally found the cause of this flaky, it's effectively related to the restart of the salt minion, as we though.

Explanation: Salt is checking every X seconds the liveness of a running job, sometimes it sends the request during the minion restart, so the query fails & the master assumes that the job has also failed.

We can fix this by sending the first job liveness query after a while, in order to allow the minion to restart.
This is easily doable by appending the `--timeout X` (where X is the number of second before the master send the first job liveness query) to the salt-run command.

**Comment by alexandre-allard** [comment]
Hopefully fixed in #2032
